nominal/sdk.py

```diff
@@ -392,17 +483,28 @@ def create_run(
             same ref name across runs, since checklists and templates use ref names to reference datasets. RIDs can be
             retrieved from `Dataset.rid` after getting or creating a dataset.
         """
-        datasets = datasets or {}
-        request = scout_run_api.CreateRunRequest(
-            attachments=list(attachment_rids),
-            data_sources={
+
+        def create_run_data_sources_for_data_source_type(
+                data_sources: Mapping[str, str], 
+                data_source_type: DataSourceType
+        ) -> dict[str, scout_run_api.CreateRunDataSource]:
+            return {
                 ref_name: scout_run_api.CreateRunDataSource(
-                    data_source=scout_run_api.DataSource(dataset=rid),
+                    data_source=scout_run_api.DataSource(**{data_source_type : rid}),
                     series_tags={},
                     offset=None,  # TODO(alkasm): support per-dataset offsets
                 )
-                for ref_name, rid in datasets.items()
-            },
+                for ref_name, rid in data_sources.items()
+            } if data_sources else {}
+
+        combined_data_sources = { 
+            **create_run_data_sources_for_data_source_type(datasets, "dataset"),
+            **create_run_data_sources_for_data_source_type(log_sets, "log_set"),
+        }
```

Comment:
Is there a more fluent way to do this? I thought that since we'd be supporting 4-5+ types I'd combine them somehow but the only type safety on providing arg name as a string is that it's of the literal `DataSourceType`